Mathlib/CategoryTheory/Localization/DerivabilityStructure/Basic.lean

```diff
@@ -0,0 +1,132 @@
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Localization.Resolution
+import Mathlib.CategoryTheory.GuitartExact.VerticalComposition
+
+/-!
+# Derivability structures
+
+Let `Φ : LocalizerMorphism W₁ W₂` be a localizer morphism, i.e. `W₁ : MorphismProperty C₁`,
+`W₂ : MorphismProperty C₂`, and `Φ.functor : C₁ ⥤ C₂` is a functors which maps `W₁` to `W₂`.
+Following the definition introduced by Bruno Kahn and Georges Maltsiniotis in
+[Bruno Kahn and Georges Maltsiniotis, *Structures de dérivabilité*][KahnMaltsiniotis2008],
+we say that `Φ` is a right derivability structure if `Φ` has right resolutions and
+the following 2-square is Guitart exact, where `L₁ : C₁ ⥤ D₁` and `L₂ : C₂ ⥤ D₂` are
+localization functors for `W₁` and `W₂`, and `F : D₁ ⥤ D₂` is the induced functor
+on the localized categories:
+
+```
+    Φ.functor
+  C₁   ⥤   C₂
+  |         |
+L₁|         | L₂
+  v         v
+  D₁   ⥤   D₂
+       F
+```
+
+## Implementation details
+
+In the field `guitartExact'` of the structure `LocalizerMorphism.IsRightDerivabilityStructure`,
+The condition that the square is Guitart exact is stated for the localization functors
+of the constructed categories (`W₁.Q` and `W₂.Q`).
+The lemma `LocalizerMorphism.isRightDerivabilityStructure_iff` show that it does
+not depend of the choice of the localization functors.
+
+## TODO
+
+* Construct derived functors using derivability structures
+* Define a constructor for derivability structures
+* Define the notion of left derivability structures
+* Construct the injective derivability structure in order to derive functor from
+the bounded below homotopy category in an abelian category with enough injectives
+* Construct the projective derivability structure in order to derive functor from
+the bounded above homotopy category in an abelian category with enough projectives
+* Construct the flat derivability structure on the bounded above homotopy category
+of categories of modules (and categories of sheaves of modules)
+* Define the product derivability structure and formalize derived functors of
+functors in several variables
+
+
+## References
+* [Bruno Kahn and Georges Maltsiniotis, *Structures de dérivabilité*][KahnMaltsiniotis2008]
+
+-/
+universe v₁ v₂ u₁ u₂
+
+namespace CategoryTheory
+
+open Category Localization
+
+variable {C₁ : Type u₁} {C₂ : Type u₂} [Category.{v₁} C₁] [Category.{v₂} C₂]
+  {W₁ : MorphismProperty C₁} {W₂ : MorphismProperty C₂}
+
+namespace LocalizerMorphism
+
+variable (Φ : LocalizerMorphism W₁ W₂)
+
+/-- A localizer morphism `Φ : LocalizerMorphism W₁ W₂` is a right derivability
+structure if it has right resolutions and the 2-square where the left and right functors
+are localizations functors for `W₁` and `W₂` are Guitart exact. -/
+class IsRightDerivabilityStructure : Prop where
+  hasRightResolutions : Φ.HasRightResolutions := by infer_instance
+  guitartExact' : TwoSquare.GuitartExact ((Φ.catCommSq W₁.Q W₂.Q).iso).hom
```

Comment:
```suggestion
  guitartExact' : TwoSquare.GuitartExact (Φ.catCommSq W₁.Q W₂.Q).iso.hom
```

Reply:
I cannot explain why, but removing the parentheses produce an error!